workspace/document.go

```diff
@@ -0,0 +1,29 @@
+package workspace
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Document is a file in a directory.
+type Document struct {
+	Root     string
+	Filepath string
+}
+
+// NewDocument creates a document from a filepath.
```

Comment:
Lets add a comment on what is Root and what is Filepath. It seems like Filepath is more of a relative path to the file.

Reply:
Yepp, good call.